aemeds/blocks/blogheader/blogheader.js

```diff
@@ -102,21 +118,31 @@ async function processChunk(block, blogs, searchTerms, container) {
     }
   }
 
-  container.append(...[...foundInHeader, ...foundInMeta, ...foundInContent]);
+  container.append(...[...foundInAuthor, ...foundInHeader, ...foundInMeta, ...foundInContent]);
+
   if (done) {
     unindicateSearch(block);
   } else {
     processChunk(block, blogs, searchTerms, container);
   }
 }
 
-async function handleSearch(block) {
+async function handleSearch(block, force = false) {
   const searchValue = block.querySelector('input').value;
-  const searchResults = block.querySelector('.search-results');
+  const oldSearchResults = block.querySelector('.search-results');
+
+  // eslint-disable-next-line prefer-destructuring
+  const parentElement = oldSearchResults.parentElement;
+  oldSearchResults.remove();
+
+  div({ class: 'search-results' });
+
+  const searchResults = div({ class: 'search-results' });
+  parentElement.appendChild(searchResults);
```

Comment:
Asking to understand, not because I have an opinion of how it should be done: Why do we have this new code and what does it do differently than this `searchResults.innerHTML= '';` below https://github.com/servicenow-martech/aemeds/pull/6/files#diff-c324584594df638df5331db1fd8349b611ac6d52665529eff5abf23ae4bd5139R153 .
Does it handle a different edge case? Do we need both?

Reply:
`searchResults.innerHTML= '';` is redundant now. Should be removed. 

The reason for creating a new `div` is because search results my overlap due to the fact that the `handleSearch` function is asynchronous and may result in a race condition. For instance
1. User types 'Gen` 
2. We start processing the first search by calling `handleSearch('Gen')` function. 
3. We call `processChunk` which loads a fist page and appends some results to the `searchResults`
4. In the meantime user continues to type ('Generative`)
5. We start processing the second search by calling `handleSearch('Generative')` function. 
6. We set `searchResults.innerHTML = '';` to clear the previous results (which we do not know if they have finished).
7. We call `processChunk` for the second search and results for second search start to appear. 
8. If the first search has not finished yet, it will continue to append results to the `searchResults` thus merging the two search results.